source/data-modeling/associations.txt

```diff
@@ -515,3 +515,410 @@ method to remove all embedded ``Album`` documents from the ``Band`` class:
    :language: ruby
    :start-after: # start-embedded-destroy-all
    :end-before: # end-embedded-destroy-all
+
+Association Behaviors
+---------------------
+
+When working with associations in {+odm+}, there are various behaviors that
+allow you to customize how associations are handled in your application. The following
+sections describe common association behaviors.
+
+Extensions
+~~~~~~~~~~
+
+Extensions allow you to add specific functionality to an association. You can
+define an extension on an association by specifying a block in the association
+definition, as shown in the following example:
+
+.. literalinclude:: /includes/data-modeling/association-behaviors.rb
+   :language: ruby
+   :start-after: # start-extensions
+   :end-before: # end-extensions
+   :emphasize-lines: 4-6
```

Comment:
q: are there any requirements or recommendations for extensions (what they're called, where they're defined, what they do, etc)?

Reply:
From the way I understand it, extensions are basically just methods that can be whatever you define